File: src/main/java/ch/hevs/sdi/springhellodata/PhoneBookRestAPI.java
package ch.hevs.sdi.springhellodata;

import org.springframework.web.bind.annotation.*;

import java.util.Collection;

@RestController
public class PhoneBookRestAPI {
    private PhoneBookRepository phoneBookRepository;

    public PhoneBookRestAPI(PhoneBookRepository phoneBookRepository) {
        this.phoneBookRepository = phoneBookRepository;
    }

    @GetMapping("/phonebook/pbservice")
    public Collection<PhoneBookRecord> getAllPhoneBookRecords() {
        return phoneBookRepository.findAll();
    }

    @GetMapping("/phonebook/pbservice/{name}")
    public PhoneBookRecord getPhoneBookRecordByName(@PathVariable String name) {
        return phoneBookRepository.findById(name).orElseThrow(RecordNotFoundException::new);
    }

    @PostMapping("/phonebook/pbservice")
    public void addPhoneBookRecord(@RequestBody PhoneBookRecord record) {
        phoneBookRepository.insert(record);
    }

    @PutMapping("/phonebook/pbservice/{name}")
    public void modifyPhoneBookRecord(@PathVariable String name, @RequestBody PhoneBookRecord record) {
        PhoneBookRecord existing = phoneBookRepository.findById(name).orElseThrow(RecordNotFoundException::new);
        existing.setPhone(record.getPhone());
        phoneBookRepository.save(existing);
    }

    @DeleteMapping("//phonebook/pbservice/{name}")
    public void deletePhoneBookRecord(@PathVariable String name) {
        phoneBookRepository.deleteById(name);
    }
 }

File: src/main/java/ch/hevs/sdi/springhellodata/PhoneBookRestAPI.kt
package ch.hevs.sdi.springhellodata

import org.springframework.web.bind.annotation.*

@RestController
class PhoneBookRestAPI(private val phoneBookRepository: PhoneBookRepository) {
    @GetMapping("/phonebook/pbservice")
    fun getAllPhoneBookRecords() = phoneBookRepository.findAll()

    @GetMapping("/phonebook/pbservice/{name}")
    fun getPhoneBookRecordByName(@PathVariable name: String): PhoneBookRecord = phoneBookRepository.findById(name).orElse(null) ?: throw RecordNotFoundException()

    @PostMapping("/phonebook/pbservice")
    fun addPhoneBookRecord(@RequestBody record: PhoneBookRecord) = phoneBookRepository.insert(record)

    @PutMapping("/phonebook/pbservice/{name}")
    fun modifyPhoneBookRecord(@PathVariable name: String, @RequestBody record: PhoneBookRecord) {
        val existing = phoneBookRepository.findById(name).orElse(null) ?: throw RecordNotFoundException()
        existing.phone = record.phone
        phoneBookRepository.save(existing)
    }

    @DeleteMapping("//phonebook/pbservice/{name}")
    fun deletePhoneBookRecord(@PathVariable name: String) = phoneBookRepository.deleteById(name)
}
